packages/server-side-render/src/server-side-render.js

```diff
@@ -112,9 +112,15 @@ export default function ServerSideRender( props ) {
 
 		setIsLoading( true );
 
-		let sanitizedAttributes =
-			attributes &&
-			__experimentalSanitizeBlockAttributes( block, attributes );
+		let sanitizedAttributes;
+		if ( attributes ) {
+			// Anchor attribute isn't supported on the server side and
+			// should be excluded from the parameters.
+			const { anchor, ...restAttributes } = attributes;
+			sanitizedAttributes =
+				restAttributes &&
+				__experimentalSanitizeBlockAttributes( block, restAttributes );
+		}
```

Comment:
On the server side, the `id` attribute is no longer recognized and must be explicitly removed from the parameter. Also, the `id` attribute should not be rendered on the block editor.